http-server/src/main/java/xtdb/jackson/TxDeserializer.java

```diff
@@ -0,0 +1,56 @@
+package xtdb.jackson;
+
+import clojure.lang.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import xtdb.IllegalArgumentException;
+import xtdb.tx.Ops;
+import xtdb.tx.Tx;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TxDeserializer extends StdDeserializer<Tx>  {
+
+    public TxDeserializer() {
+        super(Tx.class);
+    }
+
+    @Override
+    public Tx deserialize(com.fasterxml.jackson.core.JsonParser jp, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+        ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+        ObjectNode node = mapper.readTree(jp);
+        List<Ops> ops = new ArrayList<>();
+        LocalDateTime systemTime = null;
+        ZoneId defaultTz = null;
+
+        try {
+            if (node.has("tx_ops")) {
+                // The reason that we are fully walking the nodes is that Jsonista overwrites the standard `List` parser and
+                // something like `mapper.getTypeFactory().constructCollectionType(List.class, Ops.class)`
+                ArrayNode txOpsNode = (ArrayNode)  node.get("tx_ops");
+                for (JsonNode txOp: txOpsNode) {
+                    ops.add(mapper.treeToValue(txOp, Ops.class));
+                }
+            } else {
+                throw IllegalArgumentException.create(Keyword.intern("tx", "missing-tx-ops"), PersistentHashMap.EMPTY);
+            }
+            if (node.has("system_time")) {
+                systemTime = (LocalDateTime) mapper.readValue(node.get("system_time").traverse(mapper), Object.class);
+            }
+            if (node.has("default_tz")) {
+                defaultTz = (ZoneId)  mapper.readValue(node.get("default_tz").traverse(mapper), Object.class);
+            }
+        } catch (Exception e) {
```

Comment:
the exception handling is a little strange here - you're throwing IAE on L42 but then catching and ignoring it here - let's pair up and talk about the wider strategy?

Reply:
yes, I want to keep the most inner IAE exception.

Reply:
My approach has been to write something like 
```java
  } catch (IllegalArgumentException i) {
           throw i;
  } catch (Exception e) {
           throw IllegalArgumentException.create(Keyword.intern("xtdb", "malformed-tx"), PersistentHashMap.create(Keyword.intern("json"), node.toPrettyString()));
  }
```
which I didn't do here.

Reply:
right - I wouldn't go with the `catch Exception` here - if it's an IAE, you want the inner one (as you say), maybe with some context at a push (but that's not what's happening here); if it's not an IAE (e.g. a general `IOException`) then it's likely not a malformed-tx

Reply:
As suggested below, we could use `JsonProcessingException` in place of Exception here. For instance, I don't do a `instanceof` checks everywhere. So there are definitely malformed tx/queries which won't get caught if I remove the second case.